e2e/testcases/cli_test.go

```diff
@@ -1068,3 +1068,46 @@ func TestNomosVersion(t *testing.T) {
 		nt.T.Fatalf("Expected to find config-sync component in output:\n%s\n", string(out))
 	}
 }
+
+func TestNomosStatusNameFilter(t *testing.T) {
+	nt := nomostest.New(t, nomostesting.NomosCLI)
+
+	// get status
+	cmd := nt.Command("nomos", "status", "--name", "root")
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		nt.T.Log(string(out))
+		nt.T.Fatal(err)
+	}
+
+	if strings.Contains(string(out), "bookinfo:repo-sync") {
+		nt.T.Fatalf("Expected to not find repo-sync component in output:\n%s\n", string(out))
+	}
+
+	if !strings.Contains(string(out), "<root>:root-sync") {
+		nt.T.Fatalf("Expected to find root-sync component in output:\n%s\n", string(out))
+	}
+}
+
+func TestNomosStatusNameNamespaceFilter(t *testing.T) {
```

Comment:
Using a separate test case requires resetting the test env when --share-test-env is true.
I would suggest combining them into a single test case.

Another suggestion is to test the combination of the --name and --namespace flag.

For example, in the test, you can create multiple RootSyncs and RepoSyncs with different namespaces.
Then your can verify:
- '--namespace=c-m-s' only should list all RootSyncs
- '--namespace=***' only should list all RepoSyncs in the specific namespace
- '--name=*** only' should list both RootSyncs and RepoSyncs with the same name (exact match)
- '--name=*** and --namespace=***' should list RootSyncs **OR** RepoSyncs that match with the flags.

Reply:
the testcase has been merged to 1 big testcase that setup a cluster with multiple root and reposync and perform validation for each of the 4 listed scenario